json-c function `json_util_get_last_err` not found

After doing the propper cloning setup and running make the following error mesage occurs:

## Error
```
NVME_VERSION = 1.14.89.g81b84
make -C libnvme/
make[1]: Entering directory `/root/nvme-cli/libnvme'
Makefile:34: config-host.mak: No such file or directory
Running configure ...
prefix                        /usr
includedir                    /usr/include
libdir                        /usr/lib
mandir                        /usr/man
datadir                       /usr/share
libuuid                       yes
systemd                       no
libjson-c                     yes
cpp                           yes
make[1]: Leaving directory `/root/nvme-cli/libnvme'
make[1]: Entering directory `/root/nvme-cli/libnvme'
sed -e "s%@prefix@%/usr%g" -e "s%@libdir@%/usr/lib%g" -e "s%@includedir@%/usr/include%g" -e "s%@NAME@%libnvme%g" -e "s%@VERSION@%0.1%g" libnvme.pc.in >libnvme.pc
make[2]: Entering directory `/root/nvme-cli/libnvme/src'
cc -g -fomit-frame-pointer -O2 -I/usr/include -Invme/ -I../ccan/ -include ../config-host.h -D_GNU_SOURCE -Wall -fPIC    ../ccan/tools/configurator/configurator.c   -o ../ccan/tools/configurator/configurator
../ccan/tools/configurator/configurator > ../ccan/config.h
     CC ../ccan/ccan/str/str.ol
     CC ../ccan/ccan/str/debug.ol
     CC ../ccan/ccan/list/list.ol
     CC nvme/ioctl.ol
     CC nvme/filters.ol
     CC nvme/fabrics.ol
     CC nvme/util.ol
     CC nvme/tree.ol
     CC nvme/log.ol
     CC nvme/cleanup.ol
     CC nvme/json.ol
     AR libnvme.a
ar: creating libnvme.a
 RANLIB libnvme.a
     CC nvme/ioctl.os
     CC nvme/filters.os
     CC nvme/fabrics.os
     CC nvme/util.os
     CC nvme/tree.os
     CC nvme/log.os
     CC nvme/cleanup.os
     CC nvme/json.os
     CC ../ccan/ccan/str/str.os
     CC ../ccan/ccan/str/debug.os
     CC ../ccan/ccan/list/list.os
     CC libnvme.so.1.0.1
make[2]: Leaving directory `/root/nvme-cli/libnvme/src'
make[2]: Entering directory `/root/nvme-cli/libnvme/test'
     CC test
../src//libnvme.a(json.ol): In function `json_read_config':
/root/nvme-cli/libnvme/src/nvme/json.c:155: undefined reference to `json_util_get_last_err'
../src//libnvme.a(json.ol): In function `json_update_config':
/root/nvme-cli/libnvme/src/nvme/json.c:272: undefined reference to `json_util_get_last_err'
collect2: error: ld returned 1 exit status
make[2]: *** [test] Error 1
make[2]: Leaving directory `/root/nvme-cli/libnvme/test'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/root/nvme-cli/libnvme'
make: *** [libnvme] Error 2
```

## Versions
Make: GNU Make 3.82
Kernel: 5.10.29

Installed json Libraries
```
json-c.x86_64                       0.11-4.amzn2.0.4                 @amzn2-core
json-c-devel.x86_64                 0.11-4.amzn2.0.4                 @amzn2-core
jsoncpp.x86_64                      1.7.2-3.amzn2.0.2                @amzn2-core
```

Afer that I compiled json-c by hand but this did not fix anything.

--- a/Makefile
+++ b/Makefile
@@ -42,7 +42,8 @@ ifeq ($(HAVE_SYSTEMD),0)
 endif
 
 ifeq ($(LIBJSONC), 0)
-	override LDFLAGS += -ljson-c
+	override LDFLAGS += $(shell pkg-config --libs json-c)
+	override CFLAGS += $(shell pkg-config --cflags json-c)
 	override CFLAGS += -DLIBJSONC
 endif
 

--- a/meson.build
+++ b/meson.build
@@ -31,8 +31,8 @@ libuuid = dependency('uuid', required: true)
 conf.set('LIBUUID', libuuid.found(), description: 'Is libuuid required?')
 
 # Check for libjson-c availability
-libjson = dependency('json-c', required: true)
-conf.set('LIBJSONC', libjson.found(), description: 'Is json-c required?')
+json_c = dependency('json-c', version: '>=0.13', fallback : ['json-c', 'json_c'])
+conf.set('CONFIG_JSONC', json_c.found(), description: 'Is json-c required?')
 
 # Check for libhugetlbfs  availability
 libhugetlbfs = dependency('hugetlbfs', required: false)
@@ -78,5 +78,5 @@ subdir('util')
 executable(
     'nvme',
     sources,
-    dependencies: [ libnvme_dep, libuuid, libjson ],
+    dependencies: [ libnvme_dep, libuuid, json_c ],
 )

--- a/nvme.h
+++ b/nvme.h
@@ -23,7 +23,7 @@
 
 #include "plugin.h"
 #ifdef LIBJSONC
-#include <json-c/json.h>
+#include <json.h>
 
 #define json_create_object(o) json_object_new_object(o)
 #define json_create_array(a) json_object_new_array(a)

